slash-commands/moderation/records.js

```diff
@@ -0,0 +1,58 @@
+const {SlashCommandBuilder} = require('@discordjs/builders');
+const {notesPresenter} = require('../../presenters/dataPresenters');
+const {infractionsPresenter} = require('../../presenters/dataPresenters');
+const {displayRecordsLog} = require('../../presenters/logPresenters');
+const {promisePool} = require('../../config/db');
+
+module.exports = {
+  data: new SlashCommandBuilder()
+    .setName('records')
+    .setDescription('Finds all user records in db and returns them to channel')
+    .addUserOption((option) =>
+      option
+        .setName('target')
+        .setDescription('The user (mention or id)')
+        .setRequired(true)
+    ),
+
+  async execute(interaction) {
+    try {
+      const targetUser = await interaction.options.getUser('target');
+      const notes = await getNotesFromDB(targetUser);
+      const infractions = await getInfractionsFromDB(targetUser);
+      await displayRecordsLog({
+        interaction: interaction,
+        targetUser: targetUser,
+        recordCollections: [
+          infractionsPresenter(infractions),
+          notesPresenter(interaction, notes),
+        ],
+      });
+    } catch (err) {
+      console.error(err);
+      interaction.reply(`There was an error reading the records.`);
+    }
+  },
+};
+
+async function getInfractionsFromDB(targetUser) {
+  try {
+    const sqlInfractions = `SELECT timestamp,reason,id,action,valid FROM infractions WHERE user = ?`;
+    const [infractions] = await promisePool.execute(sqlInfractions, [
+      targetUser.id,
+    ]);
+    return infractions;
+  } catch (err) {
+    throw new Error(err.message);
+  }
+}
+
+async function getNotesFromDB(targetUser) {
+  try {
+    const sqlNotes = `SELECT timestamp,moderator,id,note,valid FROM user_notes WHERE user = ?`;
+    const [notes] = await promisePool.execute(`${sqlNotes}`, [targetUser.id]);
+    return notes;
+  } catch (err) {
+    throw new Error(err.message);
+  }
+}
```

Comment:
I considered this when doing the refactor, but decided against it.  I think it would ultimately be premature optimization for very little gain.  

Each of these functions is only written/used twice in the whole app.  Since these are DB queries, I'd be hesitant to refactor them out unless we were using them in several more places.  It will also make it more difficult to change in the future if we want to run a different query for notes/infractions vs. for records if we move this to a separate, shared file.

I say we keep it as is and consider a further refactor down the line if we ever need these in more than the two places.

Reply:
Sounds good.